features/step_definitions/dish_steps.rb

```diff
@@ -0,0 +1,9 @@
+Given(/^the following dishes exist$/) do |table|
+  table.hashes.each do |hash|
+    rest = Restaurant.find_by(name: hash["restaurant"])
+    hash["menu"] = Menu.where(name: hash["menu"], restaurant_id: rest.id).first
+    hash["price"] = hash["price"].to_i
+    restaurant = hash.delete("restaurant")
+    FactoryGirl.create(:dish, hash)
+  end
```

Comment:
Not sure about this block. Can you try to explain each line? What is does and why?